src/client/app/effects/snippet.ts

```diff
@@ -187,6 +187,10 @@ export class SnippetEffects {
         .map((updatedSnippet) => new Snippet.SaveAction(updatedSnippet))
```

Comment:
Now that your "massageSnippet" does its own save too, do you need this one?  Or will it end up saving a sample, even though the sample shouldn't be saved?